packages/openapi/src/decorators.ts

```diff
@@ -58,19 +60,89 @@ export const $extension: ExtensionDecorator = (
   extensionName: string,
   value: TypeSpecValue,
 ) => {
+  // Convert the TypeSpec value to JSON and collect any diagnostics
+  const [data, diagnostics] = typespecTypeToJson(value, entity);
+  const isModelProperty = entity.kind === "ModelProperty";
+
+  // Handle the "uniqueItems" extension
+  if (extensionName === "uniqueItems") {
+    // Check invalid target
+    if (
+      !(
+        isModelProperty &&
+        entity.type.kind === "Model" &&
+        isArrayModelType(context.program, entity.type)
+      )
+    ) {
+      // Report diagnostic if the target is invalid for "uniqueItems"
+      reportDiagnostic(context.program, {
+        code: "invalid-extension-target",
+        messageId: "uniqueItems",
+        format: { paramName: entity.kind },
+        target: entity,
+      });
+      return;
+    }
+
+    // Check invalid data, Report diagnostic if the extension value is not a boolean
+    if (data !== true && data !== false) {
+      reportDiagnostic(context.program, {
+        code: "invalid-extension-value",
+        messageId: "uniqueItems",
+        format: { extensionName: extensionName },
+        target: entity,
+      });
+      return;
+    }
+    // Set the extension for "uniqueItems"
+    setExtension(context.program, entity, extensionName, data);
+    return;
+  }
+
+  // Handle other schema extensions
+  if (schemaExtensions.includes(extensionName) && extensionName !== "uniqueItems") {
```

Comment:
`multipleof` only applies to `number`: https://www.learnjsonschema.com/draft4/validation/multipleof/